### PR TITLE
Http gap 3359 v2

### PR DESCRIPTION
**Pull request by catenacyber**
Needs https://github.com/OISF/suricata/pull/4951 and https://github.com/OISF/libhtp/pull/296

1. Fixes http sticky server
There should be 29 alerts and not only 28 (as you can count in the pcap with `grep -a Omniture http-sticky-server.pcap | wc -l`
The missed alert is at packet 8363 (cf Wireshark filter `tcp.port == 56401`), due to missing first request packet
2. Adds multiple tests with different HTTP GAP scenarios from https://github.com/OISF/suricata/pull/4951#issuecomment-630782701

* gap within the body ✅ 
* multiple different gaps in the same body ✅ 
* gap that starts within body but extends beyond it ✅ 
* gap that is exactly the body size ✅ 
* gap after CL header but before end of headers? : not supported by suricata PR so far
* methods: GET/HEAD, POST/PUT, etc : irrelevant ?
* etc.

### CODE REVIEW

**Comment by catenacyber** [comment]
Follows #234 by adding more tests

File: tests/http-gap-beyond-body/test.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        event_type: http
+        http.url: "/2"
+# response is not parsed in gap
+#       http.status: 200
+  - filter:
+      count: 1
+      match:
+        event_type: http
+        http.url: "/3"
+# response is no longer parsed after gap
```
Comment (catenacyber) [inline tests/http-gap-beyond-body/test.yaml:31]: For now, we only recover from gaps with known size.

In the future, we could try to recover the 3rd response even if we missed the content-length of the second one

File: tests/http-gap-whole-body/test.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      match:
+        event_type: fileinfo
+        fileinfo.size: 14
+        fileinfo.state: "CLOSED"
+        fileinfo.gaps: false
+  - filter:
+      count: 1
+      match:
+        event_type: fileinfo
+        fileinfo.size: 0
```
Comment (catenacyber) [inline tests/http-gap-whole-body/test.yaml:42]: Maybe this is not the best behavior and we would rather like `fileinfo.size: 14`
Thoughts ?

**Comment by victorjulien** [comment]
Can you split things so the GAP'ed flow is in its own test? This way we can limit that to 6.0.0 and keep the other test from failing on 5 and 4.1

**Comment by catenacyber** [comment]
Done something in #277 
Is it good for you ?

**Pull request closed without merging**